numpy/core/tests/test_numeric.py

```diff
@@ -991,9 +998,110 @@ class C(np.ndarray):
                         assert_(type(nzx_i) is np.ndarray)
                         assert_(nzx_i.flags.writeable)
 
-    # Tests that the array method
-    # call works
+    def test_count_nonzero_axis(self):
+        # Basic check of functionality
+        m = np.array([[0, 1, 7, 0, 0], [3, 0, 0, 2, 19]])
+
+        expected = np.array([1, 1, 1, 1, 1])
+        assert_equal(np.count_nonzero(m, axis=0), expected)
+
+        expected = np.array([2, 3])
+        assert_equal(np.count_nonzero(m, axis=1), expected)
+
+        self.assertRaises(ValueError, np.count_nonzero, m, axis=(1, 1))
+        self.assertRaises(TypeError, np.count_nonzero, m, axis='foo')
+        self.assertRaises(ValueError, np.count_nonzero, m, axis=3)
+        self.assertRaises(TypeError, np.count_nonzero,
```

Comment:
I see a couple more usages of ``TestCase`` test methods in this file that were already there, but the majority of code uses the preferred functions from ``numpy.testing``. So ``assert_raises`` here.